compiler-plugin/src/main/java/io/ballerina/wso2/apim/catalog/utils/CommonUtils.java

```diff
@@ -0,0 +1,122 @@
+        });
+    }
+    public static boolean diagnosticContainsErrors(SourceModifierContext context) {
+        return context.compilation().diagnosticResult()
+                .diagnostics().stream()
+                .anyMatch(d -> DiagnosticSeverity.ERROR.equals(d.diagnosticInfo().severity()));
+    }
+
+    public static AnnotationNode getServiceCatalogConfigAnnotation(String openApiDefinition) {
+        String configIdentifierString = Constants.CATALOG + SyntaxKind.COLON_TOKEN.stringValue() +
+                Constants.SERVICE_CATALOG_METADATA_ANNOTATION_IDENTIFIER;
+        IdentifierToken identifierToken = NodeFactory.createIdentifierToken(configIdentifierString);
+        Token atToken = NodeFactory.createToken(SyntaxKind.AT_TOKEN);
+        SimpleNameReferenceNode nameReferenceNode = NodeFactory.createSimpleNameReferenceNode(identifierToken);
+        MappingConstructorExpressionNode annotValue = getAnnotationExpression(openApiDefinition);
+        return NodeFactory.createAnnotationNode(atToken, nameReferenceNode, annotValue);
+    }
+
+    public static MappingConstructorExpressionNode getAnnotationExpression(String openApiDefinition) {
+        Token openBraceToken = NodeFactory.createToken(SyntaxKind.OPEN_BRACE_TOKEN);
+        Token closeBraceToken = NodeFactory.createToken(SyntaxKind.CLOSE_BRACE_TOKEN);
+        SpecificFieldNode specificFieldNode = createOpenApiDefinitionField(openApiDefinition);
+        SeparatedNodeList<MappingFieldNode> separatedNodeList = NodeFactory.createSeparatedNodeList(specificFieldNode);
+        return NodeFactory.createMappingConstructorExpressionNode(openBraceToken, separatedNodeList, closeBraceToken);
+    }
+
+    public static SpecificFieldNode createOpenApiDefinitionField(String openApiDefinition) {
+        IdentifierToken fieldName = AbstractNodeFactory.createIdentifierToken(Constants.OPEN_API_DEFINITION_FIELD);
+        Token colonToken = AbstractNodeFactory.createToken(SyntaxKind.COLON_TOKEN);
+        String encodedValue = Base64.getEncoder().encodeToString(openApiDefinition.getBytes(Charset.defaultCharset()));
+        ExpressionNode expressionNode = NodeParser.parseExpression(
+                String.format("base64 `%s`.cloneReadOnly()", encodedValue));
+        return NodeFactory.createSpecificFieldNode(null, fieldName, colonToken, expressionNode);
+    }
+
+    public static boolean isServiceCatalogConfigAnnotationAvailable(AnnotationNode annotationNode) {
+        if (!(annotationNode.annotReference() instanceof QualifiedNameReferenceNode)) {
+            return false;
+        }
+        QualifiedNameReferenceNode referenceNode = ((QualifiedNameReferenceNode) annotationNode.annotReference());
+        if (!Constants.CATALOG.equals(referenceNode.modulePrefix().text())) {
+            return false;
+        }
+        return Constants.SERVICE_CATALOG_METADATA_ANNOTATION_IDENTIFIER.equals(referenceNode.identifier().text());
```

Comment:
```suggestion
        return Constants.CATALOG.equals(referenceNode.modulePrefix().text() && Constants.SERVICE_CATALOG_METADATA_ANNOTATION_IDENTIFIER.equals(referenceNode.identifier().text());
```

Reply:
I think the current solution is more readable